### ISSUE TITLE
Exchange Vertical OrderBook Centering Issues

### ISSUE THREAD

**Issue by startailcoon** [bitshares/bitshares-ui#1689]
**Describe the bug**
The vertical orderbook is at times jumpy when the centering isn't updated properly. There is at times a timedelay when the orderbook is populated before the centering is issued.

**To Reproduce**
1. Go to 'Exchange'
2. See error

**Expected behavior**
Centering done without visual delay

**Screenshots (LiceCap, wait for it)**
![bitshares-pricejumpy](https://user-images.githubusercontent.com/12114550/42535765-9856ccd6-8490-11e8-86c6-2961b6ecf851.gif)

**Desktop (please complete the following information):**
 - OS: W10
 - Browser Light Wallet - 180629

**Comment by startailcoon** [comment]
Any progress on this @mattbeckham?

**Comment by wmbutler** [comment]
I'd suggest that we remove the centering completely and let the user scroll determine where this is positioned. Thoughts?

**Comment by mattbeckman** [comment]
Sorry, forgot I self assigned. I started work on it, and I think we should split the bid/ask/center so that we aren't trying to center vertically across a large set of rows. Separate scroll for top/bottom. I'll get a screen capture soon.

**Comment by startailcoon** [comment]
By adding the following to the code here we can handle centering much better

https://github.com/bitshares/bitshares-ui/blob/043a2d35523e83b0204f0f31dc5c9ab1f5c59ddd/app/components/Exchange/OrderBook.jsx#L366

```
        if (!this.props.horizontal) {
            this.setState({autoScroll: true}, () => {
                this.psUpdate();
            });
        }
```

This makes this redundant in the prepending statement on line 362

**Comment by startailcoon** [comment]
@mattbeckman will you complete this issue?

I think we should just set the order book market price in the middle and have two separate columns that can be scrolled independently instead. This would remove the jumping of the  middle column with the price.


Thoughts?

**Comment by wmbutler** [comment]
@startailcoon that's the way it was before, but it doesn't make a lot of sense to scroll independently since the center is where the buy and sell meet.

**Comment by startailcoon** [comment]
Right, but we need to make sure the center stay center and doesnt jump around.

Can we achieve that?

**Comment by mattbeckman** [comment]
Only had a few intermittent evenings to hit this. I couldn't solve the jump factor consistently without splitting. Pulled out StickyTable, and I have it working, but I will need to confirm it's responsive.

**Comment by wmbutler** [comment]
@mattbeckman can you sneak this into the current sprint?

**Comment by kapeer42** [comment]
Can I claim this?

**Comment by mattbeckman** [comment]
@kapeer42 Yes, but check with @startailcoon that it's still needed. I have a branch where StickyTable was pulled out of OrderBook, but OrderBook has been refactored since then, and due to conflicts may be faster to rethink it if the issue still exists.

**Comment by startailcoon** [comment]
StickyTables are still present as before, even on the new Exchange Design, and the current issue is still present on it.

Give it your best crack @kapeer42 👍

**Comment by sschiessl-bcp** [comment]
@startailcoon 
Please make sure this is also reflected in the new Exchange Design

**Comment by startailcoon** [comment]
@kapeer42 @sschiessl-bcp As it was based on the new exchange design I've cherry picked the code for PR #1875 and merged it to its branch. However, it still seems to be a slight issue with the centering, please see my comment on https://github.com/bitshares/bitshares-ui/pull/1875#issuecomment-422914176

**Comment by kapeer42** [comment]
I'm working on the issue on latest develop

**Comment by startailcoon** [comment]
There is no specific change to that part based on the new exchange design, so I wouldn't have any issue making it work once you have it ready.

**Comment by sschiessl-bcp** [comment]
Perfect, thanks!

**Comment by sschiessl-bcp** [comment]
Perfect, thanks!
